ldk/csharp/OliveHelpsLDK/OliveHelpsLDK/Filesystem/IFilesystemService.cs

```diff
@@ -5,36 +5,125 @@
 
 namespace OliveHelpsLDK.Filesystem
 {
+    /// <summary>
+    /// Provides access to the Filesystem.
+    /// </summary>
     public interface IFilesystemService
     {
+        /// <summary>
+        /// Get a directory's contents.
+        /// </summary>
+        /// <param name="directoryPath">The absolute path of the directory.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>Task resolving with a list of files in the directory.</returns>
         Task<IList<FileInfo>> QueryDirectory(string directoryPath, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Gets file info.
+        /// </summary>
+        /// <param name="filePath">The absolute path of the file.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns>Task resolving with the file information.</returns>
         Task<FileInfo> QueryFile(string filePath, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Creates a stream receiving updates to directory contents.
+        /// </summary>
+        /// <param name="directoryPath">The absolute path of the directory.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
         IStreamingCall<FileEvent> StreamDirectory(string directoryPath, CancellationToken cancellationToken = default);
+
+        /// <summary>
+        /// Create a stream receiving updates to a specific file.
+        /// </summary>
+        /// <param name="filePath">The absolute path of the file.</param>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
         IStreamingCall<FileEvent> StreamFile(string filePath, CancellationToken cancellationToken = default);
     }
-    
+
+    /// <summary>
+    /// Contains File Information.
+    /// </summary>
     public struct FileInfo
     {
+        /// <summary>
+        /// The filename.
+        /// </summary>
         public string Name;
+
+        /// <summary>
+        /// The size of the file, in bytes.
+        /// </summary>
         public long Size;
+
+        /// <summary>
+        /// The file permissions (if applicable).
+        /// TODO: Determine what values are sent for each of MacOS and Windows.
```

Comment:
Is this a TODO for later?

Reply:
It was! We're sending over the Go FileMode format, so I'm linking to that